src/utils/multiSig/getEligibleSignees.ts

```diff
@@ -0,0 +1,142 @@
+import { ColonyRole, Id } from '@colony/colony-js';
+
+import { apolloClient } from '~apollo';
+import {
+  type ColonyUserRoleFragment,
+  GetMultiSigRolesForDomainDocument,
+  type GetMultiSigRolesForDomainQuery,
+  type GetMultiSigRolesForDomainQueryVariables,
+  type ModelColonyRoleFilterInput,
+  type User,
+} from '~gql';
+import { type EligibleSignee } from '~types/multiSig.ts';
+import { notMaybe } from '~utils/arrays/index.ts';
+
+const COLONY_ROLE_TO_FILTER_KEY = {
+  [ColonyRole.Root]: 'role_1',
+  [ColonyRole.Arbitration]: 'role_2',
+  [ColonyRole.Architecture]: 'role_3',
+  [ColonyRole.Funding]: 'role_5',
+  [ColonyRole.Administration]: 'role_6',
+};
+
+interface GetEligibleSigneesParams {
+  colonyAddress: string;
+  domainId: number;
+  requiredRoles: ColonyRole[];
+}
+
+interface SigneeEntry {
+  [userAddress: string]: EligibleSignee;
+}
+
+interface SigneesPerRole {
+  [role: number]: SigneeEntry;
+}
+
+export interface GetEligibleSigneesResult {
+  signeesPerRole: SigneesPerRole;
+  countPerRole: {
+    [role: number]: number;
+  };
+  uniqueEligibleSignees: SigneeEntry;
+}
+
+export async function getEligibleSignees({
+  colonyAddress,
+  domainId,
+  requiredRoles,
+}: GetEligibleSigneesParams): Promise<GetEligibleSigneesResult> {
+  const domainIds =
+    domainId === Id.RootDomain ? [Id.RootDomain] : [Id.RootDomain, domainId];
+  const signeesPerRole: SigneesPerRole = {};
+  const uniqueEligibleSignees: SigneeEntry = {};
+
+  const roleFilter = requiredRoles
+    .filter((role) => !!COLONY_ROLE_TO_FILTER_KEY[role])
+    .map((role) => ({ [COLONY_ROLE_TO_FILTER_KEY[role]]: { eq: true } }));
+
+  const queryFilter: ModelColonyRoleFilterInput = {
+    isMultiSig: { eq: true },
+    or: roleFilter,
+  };
+  // amplify doesn't provide it in the UI, but the syntax of or: [{att: {eq: SOMETHING}}, {att2: {eq: SOMETHING@}}] works!
+  // but sadly having 2 or statements is impossible so we need to loop through domains
+
+  await Promise.all(
+    domainIds.map(async (permissionDomainId) => {
+      const response = await apolloClient.query<
+        GetMultiSigRolesForDomainQuery,
+        GetMultiSigRolesForDomainQueryVariables
+      >({
+        query: GetMultiSigRolesForDomainDocument,
+        variables: {
+          domainId: `${colonyAddress}_${permissionDomainId}`,
+          filter: queryFilter,
+        },
+        fetchPolicy: 'cache-first',
```

Comment:
Might need to be on the lookout for this, stale data and whatnot

Reply:
huh this one is actually the default one, so I'll remove it, I think it should suffice, I will however try to invalidate queries after manage permissions for multisig roles is done

Reply:
Invalidating queries would be indeed preferable 👌  At first glance, I thought this data gets updated often so I was gonna mention `cache-and-network` might be more suitable. But looking at it again, doesn't look like it's the case!